Mathlib/MeasureTheory/Measure/SeparableMeasureSpace.lean

```diff
@@ -0,0 +1,443 @@
+              (⋃ n ∈ (Finset.range (N + 1)), g ↑n))).toReal := by
+                rw [← toReal_add (measure_ne_top _ _) (measure_ne_top _ _)]
+                exact toReal_mono (add_ne_top.2 ⟨measure_ne_top _ _, measure_ne_top _ _⟩) <|
+                  measure_union_le _ _
+          _ < ε := by
+                rw [← add_halves ε]
+                apply _root_.add_lt_add
+                · rw [measure_diff (h_fin := measure_ne_top _ _),
+                    toReal_sub_of_le (ha := measure_ne_top _ _)]
+                  apply lt_of_le_of_lt (sub_le_dist ..)
+                  simp only [Finset.mem_range, Nat.lt_add_one_iff]
+                  exact (dist_comm (α := ℝ) .. ▸ hN N (le_refl N))
+                  exact (measure_mono <| iUnion_subset <|
+                    fun i ↦ iUnion_subset (fun _ ↦ subset_iUnion f i))
+                  exact iUnion_subset <| fun i ↦ iUnion_subset (fun _ ↦ subset_iUnion f i)
+                  exact MeasurableSet.biUnion (countable_coe_iff.1 inferInstance)
+                    (fun n _ ↦ (hf n).1)
+                · calc
+                    (μ ((⋃ n ∈ (Finset.range (N + 1)), f n) ∆
+                    (⋃ n ∈ (Finset.range (N + 1)), g ↑n))).toReal
+                      ≤ (μ (⋃ n ∈ (Finset.range (N + 1)), f n ∆ g n)).toReal :=
+                          toReal_mono (measure_ne_top _ _) (measure_mono biSup_symmDiff_biSup_le)
+                    _ ≤ ∑ n in (Finset.range (N + 1)), (μ (f n ∆ g n)).toReal := by
+                          rw [← toReal_sum (fun _ _ ↦ measure_ne_top _ _)]
+                          exact toReal_mono (ne_of_lt <| sum_lt_top fun _ _ ↦ measure_ne_top μ _)
+                            (measure_biUnion_finset_le _ _)
+                    _ < ∑ n in (Finset.range (N + 1)), (ε / (2 * (N + 1))) :=
+                          Finset.sum_lt_sum (fun i _ ↦ le_of_lt (hg i)) ⟨0, by simp, hg 0⟩
+                    _ ≤ ε / 2 := by
+                          simp only [Finset.sum_const, Finset.card_range, nsmul_eq_mul,
+                            Nat.cast_add, Nat.cast_one]
+                          rw [div_mul_eq_div_mul_one_div, ← mul_assoc, mul_comm ((N : ℝ) + 1),
+                            mul_assoc]
+                          exact mul_le_of_le_one_right (by linarith [ε_pos]) <|
+                            le_of_eq <| mul_one_div_cancel <| Nat.cast_add_one_ne_zero _
+    rintro - ε ε_pos
+    rcases this.2 ε ε_pos with ⟨t, t_mem, hμst⟩
+    exact ⟨t, t_mem, (lt_ofReal_iff_toReal_lt (measure_ne_top _ _)).2 hμst⟩
+
+/-- If a measure space `X` is generated by an algebra of sets which contains a monotone countable
+family of sets with finite measure spanning `X` (thus the measure is `σ`-finite), then this algebra
+of sets is measure-dense. -/
+theorem measureDense_of_generateFrom_setAglebra_of_sigmaFinite (h𝒜 : IsSetAlgebra 𝒜)
+    (S : ℕ → Set X) (monoS : Monotone S) (mS : ∀ n, MeasurableSet (S n))
+    (hμS : ∀ n, μ (S n) ≠ ∞) (hUS : ⋃ n, S n = univ) (h : ∀ n, S n ∈ 𝒜)
```

Comment:
It looks like you could use [MeasureTheory.Measure.FiniteSpanningSetsIn](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Measure/Typeclasses.html#MeasureTheory.Measure.FiniteSpanningSetsIn) (the sequence of sets there is not monotone but you can take the union of the n first sets).

Reply:
Thank you for the tip!